examples/multigpu_advtrain/manager.py

```diff
@@ -0,0 +1,118 @@
+import tensorflow as tf
+
+from cleverhans.utils import AccuracyReport
+from cleverhans.utils_mnist import data_mnist
+import svhn_input
+import cifar_input
+
+from make_model import make_model
+from evaluator import Evaluator
+
+_data_path = {'svhn':  '/ssd1/datasets/svhn/',
+              'cifar10':  '/ssd1/datasets/cifar-10/'}
+
+
+def preprocess_batch(images_batch, preproc_func=None):
+    if preproc_func is None:
+        return images_batch
+
+    with tf.variable_scope('preprocess'):
+        images_list = tf.split(images_batch, int(images_batch.shape[0]))
+        result_list = []
+        for img in images_list:
+            reshaped_img = tf.reshape(img, img.shape[1:])
+            processed_img = preproc_func(reshaped_img)
+            result_list.append(tf.expand_dims(processed_img, axis=0))
+        result_images = tf.concat(result_list, axis=0)
+    return result_images
+
+
+class Manager(object):
```

Comment:
docstring explaining what this class is / what it does.

I'm a little bit confused about why it has init and eval methods but doesn't seem to have train methods.

Reply:
This class is now turned into TrainManager and TrainerMultiGPU and TrainerSingleGPU with train and eval functions. The refactoring is done in commits between 4bd545ec41930128bfbc397fd34699e3c9e2c0e2 and f75fc0d7d4c361685578685f302a522bbd4fb2fa.